fast64_internal/oot/oot_actor.py

```diff
@@ -1,47 +1,410 @@
-
-import math, os, bpy, bmesh, mathutils
-from bpy.utils import register_class, unregister_class
-from io import BytesIO
+import bpy
 
 from ..f3d.f3d_gbi import *
 from .oot_constants import *
 from .oot_utility import *
-
 from ..utility import *
+from .oot_operators import *
 
-class OOT_SearchActorIDEnumOperator(bpy.types.Operator):
-	bl_idname = "object.oot_search_actor_id_enum_operator"
-	bl_label = "Select Actor ID"
-	bl_property = "actorID"
-	bl_options = {'REGISTER', 'UNDO'} 
+# General classes and functions
 
-	actorID : bpy.props.EnumProperty(items = ootEnumActorID, default = "ACTOR_PLAYER")
-	actorUser : bpy.props.StringProperty(default = "Actor")
+class OOT_UpgradeActors(bpy.types.Operator):
+	bl_idname = 'object.oot_upgrade_actors'
+	bl_label = "Upgrade Data Now!"
+	bl_options = {'REGISTER', 'UNDO', 'PRESET'}
+
+	version: bpy.props.IntProperty(name="OOT_ObjectProperties Version", default=0)
+	cur_version = 1 # version after property migration
```

Comment:
`version` is unused and I think this shouldn't duplicate `cur_version = 1` but rather reuse `OOT_ObjectProperties.cur_version`
(even simpler just change this operator's execute to simply call `OOT_ObjectProperties.upgrade_changed_props`)

Reply:
I tried but it gave me an error since I need to do something like ``from .oot_level import OOT_ObjectProperties`` so I figured copy pasting the code was the easiest solution (and since I can't test this it was the best thing to do tbh)

Reply:
an import is definitely better than having to update a value in two places

and yeah imports in the whole codebase are a mess. they need cleaning up anyway, dw about adding imports that may look weird